<?xml version="1.0"?>
<robot  name="gripper_robot">

<!-- only movable links have number in the name, like link1, link2 -->
<!-- only movable links have number in the name, like joint1, joint2 -->

<!-- world link -->
<link name="world"/>

<!-- base_link and its fixed joint -->
<joint name="joint_fix" type="fixed">
	<parent link="world"/>
	<child link="base_link"/>
</joint>

<link name="base_link">
	<collision>
		<origin xyz="0 0 0.5" rpy="0 0 0"/>
		<geometry>
			<box size="0.2 0.2 0.7"/>
		</geometry>
	</collision>

	<visual>
		<origin xyz="0 0 0.5" rpy="0 0 0"/>
		<geometry>
			<box size="0.2 0.2 1"/>
		</geometry>
	</visual>

	<inertial>
		<origin xyz="0 0 0.5" rpy="0 0 0"/>
		<mass value="1"/>
		<inertia
			ixx="1.0" ixy="0.0" ixz="0.0"
			iyy="1.0" iyz="0.0"
			izz="1.0"/>
	</inertial>
</link>

<!-- link1 and its revolute joint1 relative to base_link -->
<!-- rotating along z axis -->
<joint name="joint1" type="revolute">
	<parent link="base_link"/>
	<child link="link1"/>
	<origin xyz="0 0 1" rpy="0 0 0"/>
	<axis xyz="0 0 1"/>
	<limit effort="100" velocity="100" lower="-1.5708" upper="1.5708"/>
</joint>

<link name="link1">
	<collision>
		<origin xyz="0 0 0.05" rpy="0 0 0"/>
		<geometry>
			<cylinder length="0.1" radius="0.1"/>
		</geometry>
	</collision>

	<visual>
		<origin xyz="0 0 0.05" rpy="0 0 0"/>
		<geometry>
			<cylinder length="0.1" radius="0.1"/>
		</geometry>
	</visual>

	<inertial>
		<origin xyz="0 0 0.05" rpy="0 0 0"/>
		<mass value="1"/>
		<inertia
			ixx="0.001" ixy="0.0" ixz="0.0"
			iyy="0.001" iyz="0.0"
			izz="0.001"/>
	</inertial>
</link>

<gazebo reference="link1">
	<material>Gazebo/Grey</material>
</gazebo>

<!-- link2 and its revolute joint2 relative to link1 -->
<!-- rotating along y axis -->
<joint name="joint2" type="revolute">
	<parent link="link1"/>
	<child link="link2"/>
	<origin xyz="0 0 0.1" rpy="0 0 0"/>
	<axis xyz="0 1 0"/>
	<limit effort="100" velocity="100" lower="-0.1" upper="1.5708"/>
</joint>

<link name="link2">
	<collision>
		<origin xyz="0 0 0.5" rpy="0 0 0"/>
		<geometry>
			<cylinder length="1" radius="0.1"/>
		</geometry>
	</collision>

	<visual>
		<origin xyz="0 0 0.5" rpy="0 0 0"/>
		<geometry>
			<cylinder length="1" radius="0.1"/>
		</geometry>
	</visual>

	<inertial>
		<origin xyz="0 0 0.5" rpy="0 0 0"/>
		<mass value="1"/>
		<inertia
			ixx="0.1" ixy="0.0" ixz="0.0"
			iyy="0.1" iyz="0.0"
			izz="0.005"/>
	</inertial>
</link>

<gazebo reference="link2">
	<material>Gazebo/Grey</material>
</gazebo>

<!-- link3 and its revolute joint3 relative to link2 -->
<!-- rotating along y axis -->
<joint name="joint3" type="revolute">
	<parent link="link2"/>
	<child link="link3"/>
	<origin xyz="0 0 1" rpy="0 0 0"/>
	<axis xyz="0 1 0"/>
	<limit effort="100" velocity="100" lower="-0.1" upper="3.1416"/>
</joint>

<link name="link3">
	<collision>
		<origin xyz="0 0 0.5" rpy="0 0 0"/>
		<geometry>
			<cylinder length="1" radius="0.1"/>
		</geometry>
	</collision>

	<visual>
		<origin xyz="0 0 0.5" rpy="0 0 0"/>
		<geometry>
			<cylinder length="1" radius="0.1"/>
		</geometry>
	</visual>

	<inertial>
		<origin xyz="0 0 0.5" rpy="0 0 0"/>
		<mass value="1"/>
		<inertia
			ixx="0.1" ixy="0.0" ixz="0.0"
			iyy="0.1" iyz="0.0"
			izz="0.005"/>
	</inertial>
</link>

<gazebo reference="link3">
	<material>Gazebo/Grey</material>
</gazebo>

<!-- link4 and its revolute joint4 relative to link3 -->
<!-- rotating along y axis -->
<joint name="joint4" type="revolute">
	<parent link="link3"/>
	<child link="link4"/>
	<origin xyz="0 0 1" rpy="0 0 0"/>
	<axis xyz="0 1 0"/>
	<limit effort="100" velocity="100" lower="-1.5708" upper="1.5708"/>
</joint>

<link name="link4">
	<collision>
		<origin xyz="0 0 0" rpy="1.5708 0 0"/>
		<geometry>
			<cylinder length="0.4" radius="0.04"/>
		</geometry>
	</collision>

	<visual>
		<origin xyz="0 0 0" rpy="1.5708 0 0"/>
		<geometry>
			<cylinder length="0.4" radius="0.04"/>
		</geometry>
	</visual>

	<inertial>
		<origin xyz="0 0 0" rpy="1.5708 0 0"/>
		<mass value="0.4"/>
		<inertia
			ixx="0.05" ixy="0.0" ixz="0.0"
			iyy="0.002" iyz="0.0"
			izz="0.05"/>
	</inertial>
</link>

<gazebo reference="link4">
	<material>Gazebo/Red</material>
</gazebo>

<!-- link5 and its prismatic joint5 relative to link4 -->
<!-- translating along y axis -->
<joint name="joint5" type="prismatic">
	<parent link="link4"/>
	<child link="link5"/>
	<origin xyz="0 0.16 0" rpy="0 0 0"/>
	<axis xyz="0 1 0"/>
	<limit effort="100" velocity="100" lower="-0.16" upper="0.0"/>
</joint>

<link name="link5">
	<collision>
		<origin xyz="0 0 0.15" rpy="0 0 0"/>
		<geometry>
			<cylinder length="0.3" radius="0.04"/>
		</geometry>
	</collision>

	<visual>
		<origin xyz="0 0 0.15" rpy="0 0 0"/>
		<geometry>
			<cylinder length="0.3" radius="0.04"/>
		</geometry>
	</visual>

	<inertial>
		<origin xyz="0 0 0.15" rpy="0 0 0"/>
		<mass value="0.2"/>
		<inertia
			ixx="0.02" ixy="0.0" ixz="0.0"
			iyy="0.02" iyz="0.0"
			izz="0.001"/>
	</inertial>
</link>

<gazebo reference="link5">
	<material>Gazebo/Red</material>
</gazebo>

<!-- link6 and its prismatic joint6 relative to link4 -->
<!-- translating along y axis -->
<joint name="joint6" type="prismatic">
	<parent link="link4"/>
	<child link="link6"/>
	<origin xyz="0 -0.16 0" rpy="0 0 0"/>
	<axis xyz="0 1 0"/>
	<limit effort="100" velocity="100" lower="0.0" upper="0.16"/>
</joint>

<link name="link6">
	<collision>
		<origin xyz="0 0 0.15" rpy="0 0 0"/>
		<geometry>
			<cylinder length="0.3" radius="0.04"/>
		</geometry>
	</collision>

	<visual>
		<origin xyz="0 0 0.15" rpy="0 0 0"/>
		<geometry>
			<cylinder length="0.3" radius="0.04"/>
		</geometry>
	</visual>

	<inertial>
		<origin xyz="0 0 0.15" rpy="0 0 0"/>
		<mass value="0.2"/>
		<inertia
			ixx="0.02" ixy="0.0" ixz="0.0"
			iyy="0.02" iyz="0.0"
			izz="0.001"/>
	</inertial>
</link>

<gazebo reference="link6">
	<material>Gazebo/Red</material>
</gazebo>

</robot>

